prov/efa/src/efa_hmem.c

```diff
@@ -365,7 +327,7 @@ int efa_domain_hmem_validate_p2p_opt(struct efa_domain *efa_domain, enum fi_hmem
 
 	switch (p2p_opt) {
 	case FI_HMEM_P2P_REQUIRED:
-		if (!info->p2p_supported_by_device)
+		if (OFI_UNLIKELY(ofi_hmem_p2p_disabled()) || !info->p2p_supported_by_device)
```

Comment:
Shouldn't add ofi_hmem_p2p_disabled check.

FI_HMEM_SYSTEM should always support p2p.

Reply:
This function `efa_rdm_ep_set_fi_hmem_p2p_opt` is called with `EFA_HMEM_IFACE_FOREACH_NON_SYSTEM`.
This is checking the flag FI_HMEM_DISABLE_P2P.

Reply:
Good catch. I still feel we need to honor this constraint upon `fi_setopt()`.

> The `FI_HMEM_DISABLE_P2P` environment variable discussed in `fi_mr(3)` takes precedence over this setopt option

Reply:
So this function is on EP init path, after domain init - in other words,`info->p2p_supported_by_device` was already set at this point. We have 2 invariants:
- For non system HMEM, `info->p2p_supported_by_device` is always false if `ofi_hmem_p2p_disabled()` is true
- For system HMEM, `info->p2p_supported_by_device` is always true.

This means for non system HMEM `OFI_UNLIKELY(ofi_hmem_p2p_disabled()) || !info->p2p_supported_by_device` is equivalent to `!info->p2p_supported_by_device`

Reply:
Actually, Jessie's correct here. We simply return `-FI_EINVAL` for `FI_HMEM_SYSTEM` upon setopt. We need to find a way to avoid having to iterate through each iface for stuff like this in the future.